packs/tfe_fdo_nomad/templates/tfe.nomad.tpl

```diff
@@ -0,0 +1,220 @@
+      }
+      port "http" {
+        static = [[ .tfe_fdo_nomad.tfe_http_port ]]
+      }
+      port "vault" {
+        static = [[ .tfe_fdo_nomad.tfe_vault_cluster_port ]]
+      }
+    }
+
+    service {
+      name     = [[ .tfe_fdo_nomad.tfe_service_name | quote ]]
+      port     = "tfe"
+      provider = "nomad"
+
+      check {
+        type     = "http"
+        port     = "http"
+        path     = "/_health_check"
+        interval = "5s"
+        timeout  = "2s"
+      }
+    }
+
+    task "tfe-task" {
+      driver = "docker"
+
+      identity {
+         env = true
+      }
+
+       template {
+        destination = "/secrets/key.pem"
+        change_mode = "restart"
+        splay       = "60s"
+        data        = <<EOF
+{{- with nomadVar "nomad/jobs/[[ .tfe_fdo_nomad.job_name ]]" -}}
+  {{ base64Decode .key.Value }}
+{{- end -}}
+EOF
+      }
+
+      template {
+        destination = "/secrets/cert.pem"
+        change_mode = "restart"
+        splay       = "60s"
+        data        = <<EOF
+{{- with nomadVar "nomad/jobs/[[ .tfe_fdo_nomad.job_name ]]" -}}
+  {{ base64Decode .cert.Value }}
+{{- end -}}
+EOF
+      }
+      template {
+        destination = "/secrets/bundle.pem"
+        change_mode = "restart"
+        splay       = "60s"
+        data        = <<EOF
+{{- with nomadVar "nomad/jobs/[[ .tfe_fdo_nomad.job_name ]]" -}}
+  {{ base64Decode .bundle.Value }}
+{{- end -}}
+EOF
+      }
+
+      template {
+        destination = "/secrets/nomad_ca_cert.pem"
+        change_mode = "restart"
+        splay       = "60s"
+        data        = <<EOF
+{{- with nomadVar "nomad/jobs/[[ .tfe_fdo_nomad.job_name ]]" -}}
+  {{ base64Decode .nomad_ca.Value }}
+{{- end -}}
+EOF
+      }
+
+      template {
+        destination = "/secrets/nomad_cert.pem"
+        change_mode = "restart"
+        splay       = "60s"
+        data        = <<EOF
+{{- with nomadVar "nomad/jobs/[[ .tfe_fdo_nomad.job_name ]]" -}}
+  {{ base64Decode .nomad_cert.Value }}
+{{- end -}}
+EOF
+      }
+
+      template {
+        destination = "/secrets/nomad_cert_key.pem"
+        change_mode = "restart"
+        splay       = "60s"
+        data        = <<EOF
+{{- with nomadVar "nomad/jobs/[[ .tfe_fdo_nomad.job_name ]]" -}}
+  {{ base64Decode .nomad_cert_key.Value }}
+{{- end -}}
+EOF
+      }
+
+      template {
+        destination = "/secrets/secrets.env"
+        env         = true
+        change_mode = "restart"
+        data        = <<EOF
+{{- with nomadVar "nomad/jobs/[[ .tfe_fdo_nomad.job_name ]]" -}}
+TFE_LICENSE                             = {{ .tfe_license }}
+TFE_DATABASE_PASSWORD                   = {{ .db_password }}
+TFE_OBJECT_STORAGE_S3_SECRET_ACCESS_KEY = {{ .s3_secret_key }}
+TFE_REDIS_PASSWORD                      = {{ .redis_password }}
+TFE_ENCRYPTION_PASSWORD                 = {{ .tfe_encryption_password }}
+TFE_IMAGE_REGISTRY_PASSWORD             = {{ .tfe_image_registry_password }}
```

Comment:
The value shall default to TFE_LICENSE if not provided.

Reply:
@srahul3 same as mentioned for the username. Let TFE customers provide credentials mandatorily.